class-components/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+import { ICardProps } from '../../types/AppTypes';
+
+const Card: React.FC<ICardProps> = ({ result, onClick }) => {
+  const splitUrl = result.url.split('/').filter(Boolean);
+  const itemId = splitUrl[splitUrl.length - 1];

```

Comment:
```suggestion
  const itemId = useMemo(() => {
     const splitUrl = result.url.split('/').filter(Boolean);
     return splitUrl[splitUrl.length - 1];
  }, [ result ] );
```
My suggestion is to use `useMemo` hook then itemId will not be recalculated until result changes and then use `itemId` instead of `splitUrl[splitUrl.length - 1]`